sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -375,16 +373,30 @@ case class HashAggregateExec(
        """.stripMargin
     }
 
-    val codeToEvalAggFunc = if (conf.codegenSplitAggregateFunc &&
+    val codeToEvalAggFunc = {
```

Comment:
Can we refactor this and [that](https://github.com/apache/spark/pull/27019/files#diff-2eb948516b5beaeb746aadac27fbd5b4R997) to use one common function?

Reply:
Yea, I'll update.